logstash-core/spec/logstash/agent_spec.rb

```diff
@@ -296,6 +296,24 @@
     end
   end
 
+  context "--config-test" do
+    let(:cli_args) { ["-t", "-e", pipeline_string] }
+
+    context "with a good configuration" do
+      let(:pipeline_string) { "input { } filter { } output { }" }
```

Comment:
I think that, at least in this case, drying it makes it much harder to read, as the test is:

```
"testing ok config"
config = "ok config"
assertion with ok config 
```

drying would force the reader to look for ok config string somewhere in the code